packages/core/src/state-management/repository.ts

```diff
@@ -94,11 +102,168 @@ export class Repository {
    */
   #deps: RepositoryDependencies
 
+  /**
+   * Keeps track of every pinned StreamID that has had its state 'synced' (i.e. a query was sent to
+   * pubsub requesting the current tip for that stream) since the start of this process. This set
+   * only grows over time, in line with how many pinned streams get queried.
+   * @private
+   */
+  #syncedPinnedStreams: Set<string> = new Set()
+
   /**
    * Internal APIs
    */
   _internals: RepositoryInternals
 
+  /**
+   * A collection of private helper functions used by the stream loading code
+   * @private
+   */
+  readonly #loadHelpers = new (class {
```

Comment:
I'm curious what yall think about this approach.  I want to remove RepositoryInternals and fold its functionality up into Repository.  But that makes Repository very large.  All of this functionality is fairly tightly focused around stream loading, and seems like it should be possible to pull out into a self-contained module, but it isn't really actually possible to put into an entirely separate component without making a lot of the internal members of Repository public, since a lot of these functions need access to members of Repository.  Conceptually this functionality really is a part of Repository - managing the caches, state store, and network access is one of the core jobs of the Repository in my mind.  So this approach let this remain the domain of the Repository, but still have some sort of logical grouping and separation of this functionality from the rest of the public API of Repository.  It's not my favorite, but was the best I could come up with.  The alternative is just remove this subclass and put all of these functions as regular methods on Repository.

Definitely open to input or ideas here.

Reply:
If the functionality is part of the Repository, I think it's fine just having these as private methods.
The logic here seems more like a hack to me, considering there are circular references between the Repository and these internal helpers it doesn't seem to add any value over simply having these methods directly on the Repository class?

Reply:
Okay done.  I don't love how big the Repository is getting, but I'd still rather have that then the arbitrary and circular split between Repository/StateManager/RepositoryInternals.  Hopefully we can figure out some good abstractions to break Repository up at some point in the future